sdk/src/metrics/ungrouped_processor.cc

```diff
@@ -0,0 +1,173 @@
+#include "opentelemetry/sdk/metrics/ungrouped_processor.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+
+namespace sdk
+{
+
+namespace metrics
+{
+
+UngroupedMetricsProcessor::UngroupedMetricsProcessor(bool stateful)
+{
+  stateful_ = stateful;
+}
+
+/**
+ * CheckpointSelf will return a vector of records to be exported. This function will go through
+ *aggregators that have been sent through process() and return them as a vector of records.
+ **/
+std::vector<sdkmetrics::Record> UngroupedMetricsProcessor::CheckpointSelf() noexcept
+{
+  std::vector<sdkmetrics::Record> metric_records;
+
+  for (auto iter : batch_map_)
+  {
+    std::string key               = iter.first;
+    std::size_t description_index = key.find("/description/");
+    std::size_t labels_index      = key.find("/labels/");
+    std::size_t instrument_index  = key.find("/instrument/");
```

Comment:
Probably leave to TODO here for micro-optimization (instead of searching 3 strings in 3 scans, we can potentially scan once using a [DFA](https://en.wikipedia.org/wiki/Deterministic_finite_automaton)).

Reply:
Yeah this could definitely be done better, I was thinking of adding a future enhancement of using a struct to hold these values with a custom hash function instead of using this string approach. Thoughts on this?